pkg/snowflake/sql_script_test.go

```diff
@@ -0,0 +1,45 @@
+package resources_test
+
+import (
+	"database/sql"
+	"testing"
+
+	sqlmock "github.com/DATA-DOG/go-sqlmock"
+	"github.com/chanzuckerberg/terraform-provider-snowflake/pkg/provider"
+	"github.com/chanzuckerberg/terraform-provider-snowflake/pkg/resources"
+	. "github.com/chanzuckerberg/terraform-provider-snowflake/pkg/testhelpers"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/stretchr/testify/require"
+)
+
+func TestSqlScript(t *testing.T) {
+	r := require.New(t)
+	err := resources.SqlScript().InternalValidate(provider.Provider().Schema, true)
+	r.NoError(err)
+}
+
+func TestSqlScriptCreate(t *testing.T) {
+	r := require.New(t)
+
+	in := map[string]interface{}{
+		"name": "test_sql_script",
+		"lifecycle": []interface{}{
+			map[string]interface{}{
+				"create": "CREATE DATABASE good_name", // test arbitrary sql statement
+				"delete": "CREATE DATABASE good_name", // test arbitrary sql statement
+			},
+		},
+	}
+	d := schema.TestResourceDataRaw(t, resources.SqlScript().Schema, in)
+	r.NotNil(d)
+
+	WithMockDb(t, func(db *sql.DB, mock sqlmock.Sqlmock) {
+		mock.ExpectExec("CREATE DATABASE good_name").WillReturnResult(sqlmock.NewResult(1, 1))
+
+		err := resources.CreateSqlScript(d, db)
+		// The mock works when you call snowflake.Exec directly but not via
+		// resources.CreateSqlScript. Not sure why that is.
+		// err := snowflake.Exec(db, "CREATE DATABASE good_name")
+		r.NoError(err)
+	})
+}
```

Comment:
Do you also have a test for the `DeleteSqlScript`?

Reply:
Not yet, but that was the plan. I was going to write the unittest for the delete lifecycle once I got the create lifecycle test mock working.

Reply:
@alldoami I just stubbed out the [DeleteSqlScript](https://github.com/aidanmelen/terraform-provider-snowflake/blob/main/pkg/snowflake/sql_script_test.go#L47-L71)

Reply:
@alldoami Do you know why the mock is failing? I am getting an index out of bound error. It was copied 1-to-1 from the [table_test.go](https://github.com/chanzuckerberg/terraform-provider-snowflake/blob/main/pkg/resources/table_test.go#L46-L59). I think it might be failing because I have`ReadSqlScript`stubbed out to return `nil` instead of returning values from `snowflake.QueryRow(db, stmt)`